core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -284,25 +282,46 @@ object ConsumerGroupCommand extends Logging {
         case Some(-1) => LogEndOffsetResult.Unknown
         case Some(brokerId) =>
           getZkConsumer(brokerId).map { consumer =>
-            val topicAndPartition = TopicAndPartition(topicPartition.topic, topicPartition.partition)
-            val request = OffsetRequest(Map(topicAndPartition -> PartitionOffsetRequestInfo(OffsetRequest.LatestTime, 1)))
-            val logEndOffset = consumer.getOffsetsBefore(request).partitionErrorAndOffsets(topicAndPartition).offsets.head
-            consumer.close()
-            LogEndOffsetResult.LogEndOffset(logEndOffset)
+            try {
+              val (maybeErrMsg, result) = fetchLogEndOffset(topicPartition, consumer)
+              maybeErrMsg.foreach(printError(_))
+              result
+            } finally consumer.close()
           }.getOrElse(LogEndOffsetResult.Ignore)
         case None =>
           printError(s"No broker for partition '$topicPartition'")
           LogEndOffsetResult.Ignore
       }
     }
 
+    private[admin] def fetchLogEndOffset(topicPartition: TopicPartition,
+                                         consumer: SimpleConsumer): (Option[String], LogEndOffsetResult) = {
+      val topicAndPartition = TopicAndPartition(topicPartition.topic, topicPartition.partition)
+      val request = OffsetRequest(Map(topicAndPartition -> PartitionOffsetRequestInfo(OffsetRequest.LatestTime, 1)))
+      val response = consumer.getOffsetsBefore(request).partitionErrorAndOffsets(topicAndPartition)
+
+      response.error match {
+        case Errors.NONE =>
+          if (response.offsets.nonEmpty) {
+            (None, LogEndOffsetResult.LogEndOffset(response.offsets.head))
+          } else {
+            val errorMsg = s"No log end offset returned in ListOffsets response for partition $topicPartition"
+            (Some(errorMsg), LogEndOffsetResult.Ignore)
```

Comment:
The added unit test covers when there is an error in the response. Not a big deal, but should we also add one for this part of the code; i.e. when there is no error, but no offset is returned?